### PR TITLE
Split out kubelet instrumentation details into reference page

### PR DESCRIPTION
**Pull request by sftim**
Split https://kubernetes.io/docs/tasks/debug/debug-cluster/resource-metrics-pipeline/#summary-api-source away from https://kubernetes.io/docs/tasks/debug/debug-cluster/resource-metrics-pipeline/ and into a new page within the reference section.

Also, reinstate some details about `PodAndContainerStatsFromCRI`.

/sig node
/sig instrumentation
/language en
/label refactor

### CODE REVIEW

**Comment by krol3** [comment]
Hi @sftim! This PR is related to this [KEP](https://github.com/kubernetes/enhancements/issues/2371) for the release 1.26.  does this KEP have to follow the instructions for the release docs 1.26 and used the branch `dev-1.26`?
Thanks

**Comment by sftim** [comment]
As far as I know, this documentation is accurate for **Kubernetes v1.25**
The SIG Node folks should be able to confirm that detail.

After these changes land, there will be a place to make any updates we need for the v1.26 release.

**Comment by sftim** [comment]
I assure you I intend this as a change for the **v1.25** docs

/remove-milestone 1.26

**Comment by nate-double-u** [comment]
> I assure you I intend this as a change for the v1.25 docs
(sorry about that)

**Review by divya-mohan0209 — COMMENTED** [review]
Thanks Tim for this change :)
/lgtm

File: content/en/docs/reference/instrumentation/node-metrics.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+weight: 50
+description: >-
+  Mechanisms for accessing metrics at node, volume, pod and container level,
+  as seen by the kubelet.
+---
+
+The [kubelet](/docs/reference/command-line-tools-reference/kubelet/)
+gathers metric statistics at the node, volume, pod and container level,
+and emits this information in the
+[Summary API](https://github.com/kubernetes/kubernetes/blob/7d309e0104fedb57280b261e5677d919cb2a0e2d/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go).
```
Comment (tengqm) [inline content/en/docs/reference/instrumentation/node-metrics.md:13]: We can add a reference config API for this. The markdown is there, not published.

File: content/en/docs/reference/instrumentation/node-metrics.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+You can send a proxied request to the stats summary API via the
+Kubernetes API server.
+
+Here is an example of a Summary API request for a node named `minikube`:
+
+```shell
+kubectl get --raw "/api/v1/nodes/minikube/proxy/stats/summary"
+```
+
+Here is the same API call using `curl`:
```
Comment (tengqm) [inline content/en/docs/reference/instrumentation/node-metrics.md:24]: This means a kubectl proxy is started?

File: content/en/docs/reference/instrumentation/node-metrics.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+## Summary API data via CRI {#pod-and-container-stats-from-cri}
+
+{{< feature-state for_k8s_version="v1.23" state="alpha" >}}
+
+If you enable the `PodAndContainerStatsFromCRI`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) in your
+cluster, and you use a container runtime that supports statistics access via
+{{< glossary_tooltip term_id="cri" text="Container Runtime Interface">}} (CRI), then
+the kubelet fetches Pod- and container-level metric data using CRI, and not via cadvisor.
```
Comment (tengqm) [inline content/en/docs/reference/instrumentation/node-metrics.md:48]: cadvisor -> cAdvisor

File: content/en/docs/reference/instrumentation/node-metrics.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+Beginning with `metrics-server` 0.6.x, `metrics-server` queries the `/metrics/resource`
+kubelet endpoint, and not `/stats/summary`.
+{{< /note >}}
+
+## Summary metrics API source {#summary-api-source}
+
+By default, Kubernetes fetches node summary metrics data using an embedded
+[cAdvisor](https://github.com/google/cadvisor) that runs with in the kubelet.
+
+## Summary API data via CRI {#pod-and-container-stats-from-cri}
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/reference/instrumentation/node-metrics.md:40]: I know what the feature is, but we might want to change the heading we use to introduce it.

**Comment by sftim** [comment]
I fixed a couple of nits. This is ready for a review.

File: content/en/docs/reference/instrumentation/node-metrics.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/reference/instrumentation/node-metrics.md:24]: Ah, good point. It does now.

File: content/en/docs/reference/instrumentation/node-metrics.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/reference/instrumentation/node-metrics.md:13]: Sounds like a nice improvement to make.

**Review by divya-mohan0209 — COMMENTED** [review]
This LGTM! Do we need a tech sign off?

File: content/en/docs/reference/instrumentation/node-metrics.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
+---
+title: Node metrics data
```
Comment (reylejano) [inline content/en/docs/reference/instrumentation/node-metrics.md:2]: non-blocking nit
```suggestion
title: Node Metrics Data
```

Comment (sftim) [reply content/en/docs/reference/instrumentation/node-metrics.md:2]: I can make that a Good First Issue.

**Comment by reylejano** [comment]
Preview of updated pages:
- https://deploy-preview-37980--kubernetes-io-main-staging.netlify.app/docs/reference/instrumentation/node-metrics/
- https://deploy-preview-37980--kubernetes-io-main-staging.netlify.app/docs/reference/node/
- https://deploy-preview-37980--kubernetes-io-main-staging.netlify.app/docs/tasks/debug/debug-cluster/resource-metrics-pipeline/

I don't think we need a tech review since the changes in this PR are from existing documentation or in a KEP:
- the content from the new [node-metrics page](https://deploy-preview-37980--kubernetes-io-main-staging.netlify.app/docs/reference/instrumentation/node-metrics/) is from the current https://kubernetes.io/docs/tasks/debug/debug-cluster/resource-metrics-pipeline/#summary-api-source
- info about `PodAndContainerStatsFromCRI` is also stated in https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ 
- Kubernetes fetches node summary from cAdvisor is also stated in this [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2371-cri-pod-container-stats#current-state-of-these-metrics) in the section on current state of metrics: 
  > cAdvisor stats provider
  >    Calls cAdvisor directly to obtain node, pod, and container stats

/approve

**Pull request merged**